client/packages/programs/src/RnRForms/DetailView/ContentArea.tsx

```diff
@@ -1,60 +1,329 @@
+            <th>{t('label.rnr-received')}</th>
+            <th>{t('label.rnr-consumed')}</th>
+            <th>{t('label.rnr-consumed-adjusted')}</th>
+            <th>{t('label.rnr-adjustments')}</th>
+            <th>{t('label.rnr-stock-out-duration')}</th>
+            <th>{t('label.rnr-final-balance')}</th>
+            <th>{t('label.amc')}</th>
+            <th>{t('label.rnr-maximum-quantity')}</th>
+            <th>{t('label.expiry')}</th>
+            <th>{t('label.requested-quantity')}</th>
+            <th>{t('label.comment')}</th>
+            <th>{t('label.confirmed')}</th>
+          </tr>
         </thead>
 
-        {data.map((line, index) => (
-          <tr key={index}>
-            <td>{line.item.code}</td>
-            <td>{line.item.name}</td>
-            <td>{line.item.unitName}</td>
-            <td>{line.initialBalance}</td>
-            <td>{line.quantityReceived}</td>
-            <td>{line.quantityConsumed}</td>
-            <td>{line.adjustedQuantityConsumed}</td>
-            {/* <td>{line.losses}</td> */}
-            <td>{line.stockOutDuration}</td>
-            <td>{line.finalBalance}</td>
-            <td>{line.averageMonthlyConsumption}</td>
-            <td>{line.requestedQuantity}</td>
-            <td>{line.comment}</td>
-          </tr>
-        ))}
+        <tbody>
+          {lines.map(line => (
+            <RnRFormLine
+              key={line.id}
+              line={line}
+              periodLength={periodLength}
+              saveLine={saveLine}
+            />
+          ))}
+        </tbody>
       </Table>
     </Box>
   );
 };
+
+export const RnRFormLine = ({
+  line,
+  saveLine,
+  periodLength,
+}: {
+  line: RnRFormLineFragment;
+  periodLength: number;
+  saveLine: (line: RnRFormLineFragment) => Promise<void>;
+}) => {
+  const theme = useTheme();
+  const { error } = useNotification();
+
+  const [patch, setPatch] = useState<Partial<RnRFormLineFragment>>({});
+  const draft = { ...line, ...patch };
+
+  const updateDraft = (update: Partial<RnRFormLineFragment>) => {
+    const newPatch = {
+      ...patch,
+      confirmed: false,
+      ...update,
+    };
+
+    const {
+      initialBalance,
+      quantityConsumed,
+      quantityReceived,
+      adjustments,
+      stockOutDuration,
+      previousAverageMonthlyConsumption,
+    } = { ...draft, ...newPatch };
+
+    const finalBalance =
+      initialBalance + quantityReceived - quantityConsumed + adjustments;
+
+    const stockAvailableDays = periodLength - stockOutDuration;
+    const adjustedQuantityConsumed = stockAvailableDays
+      ? quantityConsumed * (periodLength / stockAvailableDays)
+      : quantityConsumed;
+
+    // This calculation might be a plugin in future!
+    const averageMonthlyConsumption =
+      // Average of the last 3 months (including the current month)
+      // TODO: what if don't want to consider previous months?
+      (2 * previousAverageMonthlyConsumption + adjustedQuantityConsumed) / 3;
```

Comment:
i.e. on first R&R form, probably DO want AMC to only consider this period...

- make AMC an editable field?
- return list of previous monthly consumption values, so if it's empty we only consider this period? ... Would we store this on the line or query it each time?